### PR TITLE
Adding users endpoints

### PR DESCRIPTION
**Pull request by JiaanNie**

### CODE REVIEW

File: src/services/users/types/userRequestTypes.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  code: string,
+  redirect_uri: string
+}
+export interface LoginRequest {
+  user_login_data: {
+    username: string,
+    password: string
+  }
+}
+
```
Comment (ghost) [inline src/services/users/types/userRequestTypes.ts:27]: Inconsistent spacing in this file

File: src/services/users/types/userResponseTypes.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  }
+  picture: string,
+  data_created: string,
+  data_updated: string,
+  date_deleted: string,
+  last_login: string,
+  uuid: string,
+}
+
+
```
Comment (ghost) [inline src/services/users/types/userResponseTypes.ts]: Inconsistent spacing again

File: src/services/users/types/userResponseTypes.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+export interface LoginResponse {
```
Comment (ghost) [inline src/services/users/types/userResponseTypes.ts:1]: Inconsistent variable naming, some are snake case some are camel case

File: src/services/users/types/userResponseTypes.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+export interface RefreshTokensResponse {
+  AccessToken: string,
+  ExpiresIn: number,
+  TokenType: string,
+  IdToken: string,
+}
+
+export interface ResetPasswordResponse {
+
+}
```
Comment (ghost) [inline src/services/users/types/userResponseTypes.ts:37]: Do we want an empty interface, can we just get rid of this?

File: src/services/users/userTypes.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+export interface createUserRequest {
```
Comment (ghost) [inline src/services/users/userTypes.ts]: This is a copy of the one in User Request types, we should try to re-use this type or something instead

File: src/services/users/userTypes.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    email: string,
+    first_name: string,
+    last_name: string,
+    phone_number: string,
+    country: string,
+    verified_legal_terms: boolean,
+  }
+}
+
+export interface authCodeRequest {
```
Comment (ghost) [inline src/services/users/userTypes.ts]: Another interface copy

File: src/services/users/users.ts [hunk 7]
```diff
@@ -1,29 +1,101 @@
...
+  refreshTokens(appClientId: string, payload: string): Promise<RefreshTokensResponse> {
+    return this.axiosClientHeaderless.post<RefreshTokensResponse>(
+      `/auth/${appClientId}/refresh-token`,
+      payload
+    ).then((res: AxiosResponse<RefreshTokensResponse>) => {
+      this.logger.debug(res);
+      return res.data
+    });
+  }
+  validateEmail(payload: ValidateEmailRequest):Promise<ValidateEmailResponse> {
```
Comment (ghost) [inline src/services/users/users.ts]: Inconsistent spacing

**Review by ghost — COMMENTED** [review]
Some changes needed, but when you fix those don't forget to post this PR in the developer channel

File: src/services/users/types/userRequestTypes.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
+import {ValidateEmailResponse} from "./userResponseTypes";
+
+export interface SignUpRequest {
```
Comment (ShawnClake) [inline src/services/users/types/userRequestTypes.ts]: `SignUp` should be `Signup`. We treat it as a single word in most other places

File: src/services/users/types/userResponseTypes.ts (hunk 3 above)
Comment (ShawnClake) [reply src/services/users/types/userResponseTypes.ts:1]: @jherasymuik-ehelply I think Wilson is correct here. These names would come from the Pydantic models that were created in users for the endpoint payloads and endpoint responses.

File: src/services/users/types/userResponseTypes.ts (hunk 4 above)
Comment (ShawnClake) [reply src/services/users/types/userResponseTypes.ts:37]: We want to have a type that corresponds to the return of every endpoint. I would be more curious whether this endpoint in users actually returns something or not. If it does, it should be reflected here, if it doesn't then users should probably be changed to return something.

File: src/services/users/userTypes.ts (hunk 5 above)
Comment (ShawnClake) [reply src/services/users/userTypes.ts]: Yeah we should use `extends` here @nie209

Comment (ShawnClake) [reply src/services/users/userTypes.ts]: Also class name should use upper pascal case

File: src/services/users/userTypes.ts (hunk 6 above)
Comment (ShawnClake) [reply src/services/users/userTypes.ts]: Class name needs to user upper pascal case

**Review by ShawnClake — CHANGES_REQUESTED** [review]
Overall, just remember to run ctrl-alt-L on your files before pushing. That would make it so @jherasymuik-ehelply doesnt complain about inconsistent spacing :)

File: src/services/users/types/userResponseTypes.ts (hunk 3 above)
Comment (ghost) [reply src/services/users/types/userResponseTypes.ts:1]: Makes sense! I'll close this comment

File: src/services/users/userTypes.ts (hunk 5 above)
Comment (JiaanNie) [reply src/services/users/userTypes.ts]: ohhh this file i forgot to remove it. all the types is in a different place. we have a type folder  within the folder that contain two file `userRequestTypes.ts` and `useResponseTypes.ts`

File: src/services/users/userTypes.ts (hunk 6 above)
Comment (JiaanNie) [reply src/services/users/userTypes.ts]: the correct interface is in `userRequestTypes.ts` file

File: src/services/users/types/userResponseTypes.ts (hunk 4 above)
Comment (JiaanNie) [reply src/services/users/types/userResponseTypes.ts:37]: the endpoint itself it return what cognito client return. it was a big hard to find out what to return. i might need a bit help with this one

Comment (ShawnClake) [reply src/services/users/types/userResponseTypes.ts:37]: This shouldn't be an issue when Jenn refactors users. All endpoints will need to return our own model and nothing directly from a cognito client

**Review by ShawnClake — APPROVED** [review]
Same as the other PR. More inconsistent spacing. I think part of the problem is that your IDE is probably configured to use an indent of 2 when we want 4 for JavaScript. I never mentioned this in the code standards, so I will add that when I get a chance.

**Pull request merged**